godtools/App/Features/ToolsFilter/Presentation/ToolFilterCategorySelectionView/ToolFilterCategorySelectionViewModel.swift

```diff
@@ -43,22 +45,37 @@ class ToolFilterCategorySelectionViewModel: ObservableObject {
         self.selectedCategory = categoryFilterSelectionPublisher.value
         self.flowDelegate = flowDelegate
         
+        getCurrentAppLanguageUseCase
+            .getLanguagePublisher()
+            .assign(to: &$appLanguage)
+        
+        $appLanguage.eraseToAnyPublisher()
+            .flatMap { appLanguage in
+                
+                return viewToolFilterCategoriesUseCase.viewPublisher(filteredByLanguageId: selectedLanguage.id, translatedInAppLanguage: appLanguage)
+            }
+            .receive(on: DispatchQueue.main)
+            .sink { viewCategoryFiltersDomainModel in
+                
+                self.allCategories = viewCategoryFiltersDomainModel.categoryFilters
```

Comment:
Capturing self here will cause a retain cycle. Will need to use capture list [weak self].